src/App.tsx

```diff
@@ -1,14 +1,70 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useEffect,
+  useState,
+  useContext,
+  useMemo,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { TodoContext } from './components/TodoContext';
+import { Filter, TodoStates } from './types/Filter';
+
+const DEFAULT_FILTER: Filter = {
+  select: TodoStates.All,
+  input: '',
+};
+
+const getFiltredTodos = (filter: Filter, todos: Todo[]): Todo[] => {
+  const { input, select } = filter;
+
+  if (!input && select === TodoStates.All) {
+    return todos;
+  }
+
+  const validatedInput = input.toLowerCase().trim();
+
+  switch (select) {
+    case TodoStates.Active:
+      return todos.filter(({ completed, title }) => (
+        title.toLowerCase().includes(validatedInput) && !completed));
```

Comment:
You can create a helper for `title.toLowerCase().includes(validatedInput)` as yo do it 3 times or `filter` just once and do these checks inside it